galsim/base.py

```diff
@@ -338,6 +338,258 @@ def getGSParams(self):
         """
         return self.SBProfile.getGSParams()
 
+    def calculateHLR(self, size=None, scale=None, centroid=None, flux_frac=0.5):
+        """Returns the half-light radius of the object.
+
+        If the profile has a half_light_radius attribute, it will just return that, but in the
```

Comment:
What about cases where the profile has already been drawn?  Could we enable a version of this that takes an already-existing image?  (a method of the image class)